src/librustdoc/clean/cfg.rs

```diff
@@ -201,6 +201,35 @@ impl Cfg {
             _ => false,
         }
     }
+
+    /// Attempt to simplify this cfg by assuming that `assume` is already known to be true, will
+    /// return `None` if simplification managed to completely eliminate any requirements from this
+    /// `Cfg`.
+    pub(crate) fn simplify_with(&self, assume: &Cfg) -> Option<Cfg> {
```

Comment:
I find it hard to understand what this does. Can you give some examples? If you're feeling adventurous you could even add some simple unit tests ;)

Reply:
There's some unit tests added, let me know if they help.

Reply:
They helped a lot, thanks!